nemo_asr/calc_rtf.py

```diff
@@ -135,65 +114,53 @@ def main():
     asr_model._prepare_for_export()
 
-    processor = extract_preprocessor(asr_model, DEVICE)
-
     input_example, input_example_length  = get_samples(WAV, audio_maxlen)
-    logging.info(f'processed example shape: {input_example.shape}')
-    logging.info(f'processed example length shape: {input_example_length}')
-    processed_example, processed_example_length = preprocess_audio(processor, input_example, DEVICE)
-    processed_example = processed_example.repeat(batch_size, 1, 1)
-    processed_example_length = processed_example_length.repeat(batch_size)
-    logging.info(f'processed example shape: {processed_example.size()}')
-    logging.info(f'processed example length shape: {processed_example_length.size()}')
-
-    profiling_context = nullcontext()
-    # if FP16:
-    if PRECISION == '16':
-        precision_context = torch.cuda.amp.autocast()
-    elif PRECISION == 'bf16':
-        precision_context = torch.cuda.amp.autocast(dtype=torch.bfloat16)
-    elif PRECISION == '32':
-        pass
-    else:
-        logging.error(f'unknown precision: {PRECISION}')
-        sys.exit(1)
-
-
-    if decoding_type == 'ctc':
-        asr_model.change_decoding_strategy(decoding_cfg=None)
+    input_example = torch.tensor(input_example).to(DEVICE)
+    input_example_length = torch.tensor(input_example_length).to(DEVICE)
+    input_example = input_example.repeat(batch_size, 1)
+    input_example_length = input_example_length.repeat(batch_size)
     
     logging.info(f"running {nbatches} batches; with {warmup_batches} batches warmup; batch_size: {batch_size}")
     rtfs=[]
     for i in range(3): # average over 3 runs
         total_time = 0
-        with profiling_context:
-            with precision_context:
-                with torch.no_grad():
-                    for i in tqdm(range(nbatches + warmup_batches)):
-
-                        start = time.time()
-                        if decoding_type == 'rnnt':
-                            enc_out, enc_len = asr_model.encoder.forward(audio_signal=processed_example, length=processed_example_length)
-                            dec_out, dec_len = asr_model.decoding.rnnt_decoder_predictions_tensor(
-                            encoder_output=enc_out, encoded_lengths=enc_len, return_hypotheses=False
-                            )
-                        else:
-                            enc_out, enc_len, greedy_predictions = asr_model.forward(processed_signal=processed_example, processed_signal_length=processed_example_length)
-                            dec_out, dec_len = asr_model.decoding.ctc_decoder_predictions_tensor(
-                            enc_out, decoder_lengths=enc_len, return_hypotheses=False
-                            )
-                        torch.cuda.synchronize()
-                        end = time.time()
-                        if i >= warmup_batches:
-                            total_time += end - start
+        with torch.cuda.amp.autocast(dtype=torch.bfloat16):
```

Comment:
hmm, do we need to remove fp16 support?

Reply:
We are not removing support, but only making sure to use ampere. This is done as we are only measuring RTF on A100.

Reply:
@Vaibhavs10 to add hardware info on leaderboard space.